Arriba/Arriba.Web/parts/Search.jsx

```diff
@@ -173,78 +163,59 @@ export default React.createClass({
         this.jsonQueryWithError(
             configuration.url + "/allCount",
             data => {
-                // Do not wipe userSelectedId if currentTable is going from `undefined` to defined.
-                // Scnario: On page load with open=something and table inferred from query.
-                var currentTable = this.state.userSelectedTable || data.content.resultsPerTable[0].tableName;
-                if (this.state.currentTable && this.state.currentTable !== currentTable) {
-                    this.setState({
-                        userTableSettings: {},
-                        userSelectedId: undefined
-                    });
-                }
                 this.setState({
-                    allCountData: data,
-                    currentTable: currentTable,
+                    counts: data.content,
                     loading: false
-                }, () => {
-                    this.getTableBasics();
-                    if (then) then();
-                });
+                }, then);
 
                 data.content.parsedQuery = data.content.parsedQuery.replace(/\[\*\]:/g, ""); // Other consumers want the [*] removed also.
                 this.mru.update(data.content.parsedQuery);
             },
             { q: this.state.query }
         );
-    },
-    getTableBasics: function () {
-        // Once a table is selected, find out the columns and primary key column for the table
-
-        // If allBasics is not ready, abort.
+    }
+    getTableSettings() {

```

Comment:
Should getTableSettings be a separate stage? Is getting localStorage synchronous, so that once we have allBasics we can get the table settings directly in getListing without needing a separate phase?

Reply:
It currently separate as there's no need to recalculate the currentTableSettings upon every additional page fetch.